### PR TITLE
Refactor analytics data store

### PR DESCRIPTION
**Pull request by atorresveiga**
Closes: #9443 

### Why
As part of the Reliability project, we are working on improving the app's performance and reducing the perceived time it takes to load. One of the ideas we are working on is introducing a timestamp as the web does and preventing redundant requests to the API. This PR is part of the timestamp introduction in the My Store screen.

This how the web looks like:
Screenshot 2023-06-21 at 11 25 43

### Description
This PR refactors the AnalyticsUpdateDataStore and introduces the Analytic Data key. The idea of the Analytic Data key is to have different timestamps for every data we need for analytics (Revenue, Visitors, Top Performer Products, etc). This approach will help to keep the analytic fetching from My Store in two different use cases and to share a common cache between My Store and the Hub screens. It also changes how the keys for storing the timestamps are created. Now the keys will use the identifier(granularity or range), the  Analytic Data key the timestamp belongs, and the site id.

### Testing instructions
TC1
1. Open the App and open a granularity (ex. Today)
2. Wait until the screen finish loading
3. Press on a different granularity (ex. This week)
4.  Wait until the screen finish loading
5. Change back to the first selection (Today)
6. Check that no request is made to the API and that data is presented from the cache (Top Performers products)
7. Change to the second selection (This week)
8. Check that no request is made to the API and that data is presented from the cache  (Top Performers products)

TC2
1. Open the App and open a granularity (ex. Today)
2. Wait until the screen finish loading
3. Press on a different granularity (ex. This week)
4. Wait until the screen finish loading
5. Change back to the first selection (Today)
6. Check that no request is made to the API and that data is presented from the cache  (Top Performers products)
7. Pull-To-Refresh and check that a call is made to the API and the data is refreshed  (Top Performers products)

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/18119390/5189794a-8379-442f-895d-27d8e1cbf773

- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/hub/sync/AnalyticsUpdateDataStore.kt [hunk 1]
```diff
@@ -47,6 +69,10 @@ class AnalyticsUpdateDataStore @Inject constructor(
         }
     }
 
+    private fun getTimeStampKey(identifier: String, analyticData: AnalyticData): String {
+        return "${selectedSite.getSelectedSiteId()}${analyticData.name}$identifier"
```
Comment (atorresveiga) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/hub/sync/AnalyticsUpdateDataStore.kt:90]: Using the selected site id in the timestamp will prevent mixing the cache information for merchants with multiple sites.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/hub/sync/AnalyticsUpdateDataStore.kt [hunk 2]
```diff
@@ -14,29 +15,50 @@ import javax.inject.Inject
...
 
     suspend fun storeLastAnalyticsUpdate(
-        selectionType: StatsTimeRangeSelection.SelectionType
-    ) = storeLastAnalyticsUpdate(selectionType.identifier)
+        selectionType: StatsTimeRangeSelection.SelectionType,
+        analyticData: AnalyticData = AnalyticData.ALL
+    ) {
+        if (analyticData == AnalyticData.ALL) {
+            AnalyticData.values().forEach { dataItem ->
+                storeLastAnalyticsUpdate(getTimeStampKey(selectionType.identifier, dataItem))
```
Comment (atorresveiga) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/hub/sync/AnalyticsUpdateDataStore.kt]: The All key will save the information for All and each independent key. The idea is that if a merchant reviews the information on the Hub screen for Week-to-Day, the This Week tab in the My Store screen can reuse that information.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/hub/sync/AnalyticsUpdateDataStore.kt (hunk 1 above)
Comment (ThomazFB) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/hub/sync/AnalyticsUpdateDataStore.kt:90]: Very well thought! Thanks for this improvement!

**Review by ThomazFB — APPROVED** [review]
Interesting approach around the `AnalyticData` parameter! Looking good! :shipit:

**Pull request merged**